config/fields/mixins/pagepicker.php

```diff
@@ -3,23 +3,35 @@
 return [
     'methods' => [
         'pagepicker' => function (array $params = []) {
-            $query = $params['query'] ?? null;
-            $model = $this->model();
-            $site  = $this->kirby()->site();
+            $query    = $params['query'] ?? null;
+            $subpages = $params['subpages'] ?? true;
+            $model    = $this->model();
+            $site     = $this->kirby()->site();
 
-            if ($query) {
-                $pages = $model->query($query, 'Kirby\Cms\Pages');
-                $self  = null;
+            if (empty($query) === false) {
+                if ($subpages === true) {
+                    if ($parent = $site->find($params['parent'] ?? null)) {
+                        $pages = $parent->children();
```

Comment:
Why the check for the `parent` option here? Shouldn't always the query be executed if it's given? I'd say this is a breaking change otherwise.

Reply:
In the first query/load, `parent` index is `null`. When navigating to the parent and subpages, the `parent` index is the `id` of the clicked page.